### PR TITLE
CAIP-2 Chain ID

### PR DESCRIPTION
**Pull request by jiexi**
## Explanation
* Use CaipChainId type for internal chain representation
  * Convert back to hex/dec format only when required
* Rename `chainId` to `caipChainId`
* Add caip helpers to controller-utils for bridging the gap between our current usage of hex/dec chainId to caipChainId

See: https://github.com/MetaMask/utils/pull/116
See: https://github.com/MetaMask/metamask-extension/pull/19991

## Changelog
**NOTE:** field change from `chainId` to `caipChainId` (or `_CHAIN_ID` to `_CAIP_CHAIN_ID`) also implies type change from `Hex` to `CaipChainId`

### `@metamask/address-book-controller`

- **BREAKING**: `this.state.addressBook` is now keyed by `CaipChainId` instead of `Hex`
- **BREAKING**: `this.state.addressBook[CaipChainId][address].chainId` field changed to `caipChainId`
- **BREAKING**: `set()` defaults the `caipChainId` arg (previously `chainId`) to `eip155:1`

### `@metamask/assets-controllers/AssetsContractController`

- **BREAKING**: constructor expects `options.chainId` to be `options.caipChainId` 
- **BREAKING**: `formatIconUrlWithProxy()` expects `params.chainId` to be `params.caipChainId`
- **BREAKING**: `isTokenDetectionSupportedForNetwork` expects `chainId` arg to be `caipChainId`

### `@metamask/assets-controllers/AssetsContractController`

- **BREAKING**: constructor expects `options.chainId` to be `options.caipChainId` 

### `@metamask/assets-controllers/NFTController`

- **BREAKING**: constructor expects `options.chainId` to be `options.caipChainId` 
- **BREAKING**: `this.state. allNftContracts[userAddress]` is now keyed by `CaipChainId` instead of `Hex`
- **BREAKING**: `this.state. allNfts[userAddress]` is now keyed by `CaipChainId` instead of `Hex`
- **BREAKING**: `updateNestedNftState()` expects `passedConfig.chainId` to be `passedConfig.caipChainId`
- **BREAKING**: `onNetworkStateChange()` expects `providerConfig.chainId` to be `providerConfig.caipChainId`
- **BREAKING**: `addNft()` expects `accountParams.chainId` to now be `accountParams.caipChainId`
- **BREAKING**: `checkAndUpdateSingleNftOwnershipStatus()` expects `accountParams.chainId` to be `accountParams.caipChainId`
- **BREAKING**: `findNftByAddressAndTokenId()` expects `chainId` arg to now be `caipChainId`
- **BREAKING**: `updateNft()` expects `chainId` arg to be `caipChainId` 
- **BREAKING**: `resetNftTransactionStatusByTransactionId()` expects `chainId` arg to be `caipChainId` 

### `@metamask/assets-controllers/NFTDetectionController`

- **BREAKING**: constructor expects `options.chainId` to be `options.caipChainId` 

### `@metamask/assets-controllers/TokenDetectionController`

- **BREAKING**: constructor expects `options.chainId` to be `options.caipChainId` 
- **BREAKING**: `onNetworkStateChange()` expects `providerConfig.chainId` to be `providerConfig.caipChainId`

### `@metamask/assets-controllers/TokenListController`

- **BREAKING**: constructor expects `options.chainId` to be `options.caipChainId` 
- **BREAKING**: `onNetworkStateChange()` expects `providerConfig.chainId` to be `providerConfig.caipChainId`
- **BREAKING**: `this.state.tokensChainsCache` is now keyed by `CaipChainId` instead of `Hex`

### `@metamask/assets-controllers/TokensRatesController`

- **BREAKING**: constructor expects `options.chainId` to be `options.caipChainId` 
- **BREAKING**: `onNetworkStateChange()` expects `providerConfig.chainId` to now be `providerConfig.caipChainId`
- **BREAKING**: `findChainSlug()` expects `chainId` arg to now be `caipChainId` 
- **BREAKING**: `this.chainId` changed to `this.caipChainId`
 

### `@metamask/assets-controllers/TokensController`

- **BREAKING**: constructor expects `options.chainId` to be `options.caipChainId` 
- **BREAKING**: `onNetworkStateChange()` expects `providerConfig.chainId` to now be `providerConfig.caipChainId`
- **BREAKING**: `this.state. allTokens` is now keyed by `CaipChainId` instead of `Hex`
- **BREAKING**: `this.state.allIgnoredTokens` is now keyed by `CaipChainId` instead of `Hex`
- **BREAKING**: `this.state.allDetectedTokens` is now keyed by `CaipChainId` instead of `Hex`
- **BREAKING**: `addDetectedTokens()` expects `detectionDetails.chainId` to now be `detectionDetails.caipChainId` 

### `@metamask/controller-utils`

- **ADDED**:  `isEthCaipChainId()` expects any value and returns true if the value is a valid CaipChainId with namespace `eip155`
- **ADDED**: `toEthCaipChainId()` expects decimal or 0x prefixed hex string and returns `CaipChainId` 
- **ADDED**: `toEthChainId()` expects CaipChainId and returns reference as decimal string 
- **ADDED**: `toEthChainIdHex()` expects CaipChainId and returns reference as 0x prefixed hex string 
- **ADDED**: `toEthChainIdInt()`expects CaipChainId and returns reference as number
- **BREAKING**: `GANACHE_CHAIN_ID` changed to `GANACHE_CAIP_CHAIN_ID`
-  **BREAKING**: `BUILT_IN_NETWORKS[networkType].chainId` field is now `caipChainId`
-  **BREAKING**: `ChainId`(hex values)  changed to `BuiltInCaipChainId` 
- **CHANGED**: `NetworkId` renamed to `InfuraNetworkId`

### `@metamask/ens-controller`

- I'm not actually certain that this controller should be changed, but I have updated it. Can write changelog after confirming we are keeping these changes or not. I don't think we should though

### `@metamask/gas-fee-controller`

- **BREAKING**: constructor expects `options.getChainId()` to be `options.getCaipChainId()`
- **BREAKING**: `onNetworkStateChange()` expects `providerConfig.chainId` to now be `providerConfig.caipChainId`
- **BREAKING**: `this.messagingSystem.call('NetworkController:getState')` expects `providerConfig.chainId` to now be `providerConfig.caipChainId`

### `@metamask/message-manager/AbstractMessageManager`

- **BREAKING**: constructor expects `options.getCurrentChainId()` to be `options.getCurrentCaipChainId()`

### `@metamask/message-manager/TypedMessageManager`

- **BREAKING**: constructor expects `options.getCurrentChainId()` to be `options.getCurrentCaipChainId()`

### `@metamask/network-controller`

- **BREAKING**: `ProviderConfig.chainId` changed to `ProviderConfig.caipChainId`
- **BREAKING**: `NetworkConfiguration.chainId` changed to `NetworkConfiguration.caipChainId`
- **BREAKING**: `this.state.providerConfig.chainId` changed to `this.state.providerConfig.caipChainId`

### `@metamask/signature-controller`

- **BREAKING**: constructor expects `options.getCurrentChainId()` to be `options.getCurrentCaipChainId()`

### `@metamask/transaction-controller`

- **BREAKING**: `Transaction.chainId` changed to `Transaction.caipChainId` changed to 
- **BREAKING**: constructor expects `options.getNetworkState()` to return to include `providerConfig.caipChainId` instead of `providerConfig.chainId`

## Checklist

- [x] I've updated the test suite for new or updated code as appropriate
- [x] I've updated documentation (JSDoc, Markdown, etc.) for new or updated code as appropriate
- [x] I've highlighted breaking changes using the "BREAKING" category above as appropriate

### CODE REVIEW

File: packages/network-controller/src/NetworkController.ts [hunk 1]
```diff
@@ -56,7 +57,7 @@ const log = createModuleLogger(projectLogger, 'NetworkController');
 export type ProviderConfig = {
   rpcUrl?: string;
   type: NetworkType;
-  chainId: Hex;
+  chainId: string;
```
Comment (jiexi) [inline packages/network-controller/src/NetworkController.ts:60]: I think we should just rename all of these to `caipChainId`. Can be pretty confusing what `chainId` is in different contexts

**Comment by jiexi** [comment]
Need to thread `CaipChainId` type from `@metamask/utils` into this

File: packages/controller-utils/src/types.ts [hunk 2]
```diff
@@ -47,31 +47,30 @@ export enum BuiltInNetworkName {
...
+  [BuiltInNetworkName.LineaMainnet]: 'eip155:59144',
 } as const;
-export type ChainId = typeof ChainId[keyof typeof ChainId];
 
 /**
  * Decimal string network IDs of built-in Infura networks, by name.
  */
-export const NetworkId = {
+export const InfuraNetworkId = {
+  // should these be eip'd?..
```
Comment (jiexi) [inline packages/controller-utils/src/types.ts:65]: no, right? this is for legacy/compatibility?

File: packages/message-manager/src/TypedMessageManager.ts [hunk 3]
```diff
@@ -108,7 +108,7 @@ export class TypedMessageManager extends AbstractMessageManager<
     }
 
     if (version === 'V3' || version === 'V4') {
-      const currentChainId = this.getCurrentChainId?.();
+      const currentChainId = this.getCurrentCaipChainId?.(); // rename this
```
Comment (jiexi) [inline packages/message-manager/src/TypedMessageManager.ts:111]: todo

File: packages/transaction-controller/src/TransactionController.ts [hunk 4]
```diff
@@ -1229,9 +1231,12 @@ export class TransactionController extends BaseController<
           new Error('No sign method defined.'),
         );
         return;
-      } else if (!chainId) {
+      } else if (!caipChainId) {
         releaseLock();
-        this.failTransaction(transactionMeta, new Error('No chainId defined.'));
+        this.failTransaction(
+          transactionMeta,
+          new Error('No caipChainId defined.'),
```
Comment (jiexi) [inline packages/transaction-controller/src/TransactionController.ts:1238]: should this error change?

File: packages/ens-controller/src/EnsController.ts
Comment (jiexi) [inline packages/ens-controller/src/EnsController.ts:1]: thoughts? I think we revert these changes

File: packages/network-controller/tests/NetworkController.test.ts [hunk 5]
```diff
@@ -5201,6 +5205,7 @@ describe('NetworkController', () => {
             },
           ),
         ).rejects.toThrow(
+          // this throws "Invalid CAIP chain ID." now. thoughts?
           new Error('Value must be a hexadecimal string, starting with "0x".'),
```
Comment (jiexi) [inline packages/network-controller/tests/NetworkController.test.ts:5209]: thoughts? this new error comes from the caip chain id being invalid

File: packages/controller-utils/src/caip.ts [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+
+/**
+ * Parses an Ethereum CAIP chain ID into a hex string.
+ *
+ * @param caipChainId - The Ethereum CAIP chain ID string.
+ * @returns the chain ID as a hex string.
+ */
+export function parseEthCaipChainIdHex(caipChainId: CaipChainId): Hex {
+  const { reference } = parseCaipChainId(caipChainId);
+  return toHex(reference);
```
Comment (jiexi) [inline packages/controller-utils/src/caip.ts:59]: this will throw `Invalid Character` if value fails to convert into `BN`. Should we catch and throw something else? This is only meant to be a convenience over the caller of this also having to call `toHex()`

File: packages/controller-utils/src/caip.test.ts [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+    });
+
+    describe('invalid CAIP chain ID', () => {
+      it('throws an error', () => {
+        expect(() => parseEthCaipChainIdHex(':123')).toThrow(
+          'Invalid CAIP chain ID.',
+        );
+        expect(() => parseEthCaipChainIdHex('eip155:ABC')).toThrow(
+          'Invalid character',
+        ); // this comes from toHex(). What should we throw instead?
```
Comment (jiexi) [inline packages/controller-utils/src/caip.test.ts:76]: same question

**Comment by jiexi** [comment]
@metamaskbot publish-preview

File: packages/controller-utils/src/caip.ts [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+}
+
+/**
+ * Parses an Ethereum CAIP chain ID into an integer.
+ *
+ * @param caipChainId - The Ethereum CAIP chain ID string.
+ * @returns the chain ID as an integer.
+ */
+export function parseEthCaipChainIdInt(caipChainId: CaipChainId): number {
```
Comment (jiexi) [inline packages/controller-utils/src/caip.ts:68]: alternative names for these helpers could be

```
toEthCaipChainId
toEthChainId
toEthChainIdHex
toEthChainIdInt
```

which would be done to match `toHex`

File: packages/network-controller/src/create-auto-managed-network-client.test.ts [hunk 9]
```diff
@@ -18,7 +18,7 @@ describe('createAutoManagedNetworkClient', () => {
     {
       type: NetworkClientType.Custom,
       rpcUrl: 'https://test.chain',
-      chainId: '0x1337',
+      caipChainId: 'eip155:1337',
```
Comment (BelfordZ) [inline packages/network-controller/src/create-auto-managed-network-client.test.ts:21]: should this be the decimal form of 0x1337? (4919)

Comment (BelfordZ) [reply packages/network-controller/src/create-auto-managed-network-client.test.ts:21]: oh wait, this one actually was already in base 10, but also 0x prefixed. the hex form should have been 0x539. RIP

File: packages/controller-utils/src/caip.ts (hunk 8 above)
Comment (BelfordZ) [reply packages/controller-utils/src/caip.ts:68]: Agreed that there could be some better naming on these conversion helpers. 

Second, should we ever support turning a chainId into an actual integer? Im not sure its 100% clear wether max chainId fits safely in an integer. It also might get confusing if some places have a base10 string, some places have a base10 integer... can we consolidate the format into being always a string? maybe thats outside the scope of this PR, just putting it out there though.

File: packages/controller-utils/src/types.ts (hunk 2 above)
Comment (BelfordZ) [reply packages/controller-utils/src/types.ts:65]: Correct, shouldnt touch these.

File: packages/assets-controllers/src/AssetsContractController.test.ts [hunk 10]
```diff
@@ -59,7 +58,7 @@ async function setupControllers() {
   });
   const preferences = new PreferencesController();
   const assetsContract = new AssetsContractController({
-    chainId: ChainId.mainnet,
+    caipChainId: SupportedTokenDetectionNetworks.mainnet,
```
Comment (adonesky1) [inline packages/assets-controllers/src/AssetsContractController.test.ts:61]: The values in [SupportedTokenDetectionNetworks](https://github.com/MetaMask/core/blob/18b64149a46b911d5f26b238e797b163374b2c08/packages/assets-controllers/src/assetsUtil.ts#L143) don't appear to be caip formatted?

Comment (adonesky1) [reply packages/assets-controllers/src/AssetsContractController.test.ts:61]: oh whoops I see they are modified in this PR.

File: packages/assets-controllers/src/AssetsContractController.ts [hunk 11]
```diff
@@ -82,19 +85,19 @@ export class AssetsContractController extends BaseController<
    * Creates a AssetsContractController instance.
    *
    * @param options - The controller options.
-   * @param options.chainId - The chain ID of the current network.
+   * @param options.caipChainId - The chain ID of the current network.
```
Comment (adonesky1) [inline packages/assets-controllers/src/AssetsContractController.ts:88]: [nit]
```suggestion
   * @param options.caipChainId - The CAIP formatted chain ID of the current network.
```

**Comment by adonesky1** [comment]
Needs to be a breaking changelog entry indicating that `SupportedTokenDetectionNetworks` is now CAIP formatted

File: packages/controller-utils/src/caip.test.ts [hunk 12]
```diff
@@ -0,0 +1,98 @@
...
+    expect(isEthCaipChainId('eip155:5')).toBe(true);
+    expect(isEthCaipChainId('eip155:1337')).toBe(true);
+    expect(isEthCaipChainId('bip122:1')).toBe(false);
+    expect(isEthCaipChainId('eip1555:1')).toBe(false);
+    expect(isEthCaipChainId('eip:1')).toBe(false);
+    expect(isEthCaipChainId('eip:')).toBe(false);
+    expect(isEthCaipChainId('eip155')).toBe(false);
+  });
+
+  describe('buildEthCaipChainId', () => {
```
Comment (adonesky1) [inline packages/controller-utils/src/caip.test.ts:21]: If passed a valid ethCaipChainId will this function just return the arg unmodified?

File: packages/controller-utils/src/caip.test.ts (hunk 7 above)
Comment (adonesky1) [reply packages/controller-utils/src/caip.test.ts:76]: "Invalid CAIP chainId"?

File: packages/controller-utils/src/caip.test.ts [hunk 13]
```diff
@@ -0,0 +1,98 @@
...
+        expect(parseEthCaipChainIdInt('eip155:1337')).toBe(1337);
+      });
+    });
+
+    describe('invalid CAIP chain ID', () => {
+      it('returns NaN', () => {
+        const invalidChainId = parseEthCaipChainIdInt(':123');
+        expect(Number.isNaN(invalidChainId)).toBe(true);
+        const invalidChainIdReference = parseEthCaipChainIdInt('eip155:ABC');
+        expect(Number.isNaN(invalidChainIdReference)).toBe(true);
```
Comment (adonesky1) [inline packages/controller-utils/src/caip.test.ts:94]: Should we throw here instead?

File: packages/controller-utils/src/caip.ts [hunk 14]
```diff
@@ -0,0 +1,75 @@
...
+ */
+export function buildEthCaipChainId(chainId = ''): CaipChainId {
+  const chainIdDecimal = chainId.startsWith('0x')
+    ? parseInt(chainId, 16).toString(10)
+    : chainId;
+
+  if (Number.isNaN(parseInt(chainIdDecimal, 10))) {
+    throw new Error(`Invalid chain ID "${chainId}"`);
+  }
+  return `eip155:${chainIdDecimal}` as CaipChainId;
```
Comment (adonesky1) [inline packages/controller-utils/src/caip.ts:37]: If passed a valid ethCaipChainId should we simply return the arg unmodified?

File: packages/controller-utils/src/caip.ts (hunk 6 above)
Comment (adonesky1) [reply packages/controller-utils/src/caip.ts:59]: Yeah I think 'Invalid Caip ChainId'?

File: packages/controller-utils/src/constants.ts [hunk 15]
```diff
@@ -126,12 +130,12 @@ export enum ApprovalType {
...
-  [NetworkId.goerli]: NetworkType.goerli,
-  [NetworkId.sepolia]: NetworkType.sepolia,
-  [NetworkId.mainnet]: NetworkType.mainnet,
-  [NetworkId['linea-goerli']]: NetworkType['linea-goerli'],
-  [NetworkId['linea-mainnet']]: NetworkType['linea-mainnet'],
+  [InfuraNetworkId.goerli]: NetworkType.goerli,
+  [InfuraNetworkId.sepolia]: NetworkType.sepolia,
+  [InfuraNetworkId.mainnet]: NetworkType.mainnet,
+  [InfuraNetworkId['linea-goerli']]: NetworkType['linea-goerli'],
+  [InfuraNetworkId['linea-mainnet']]: NetworkType['linea-mainnet'],
```
Comment (adonesky1) [inline packages/controller-utils/src/constants.ts:140]: What was the reason for this change? How does it relate to the CAIP-2 migration?

Comment (adonesky1) [reply packages/controller-utils/src/constants.ts:140]: Oh I see the name just changed for the type

File: packages/ens-controller/src/EnsController.ts [hunk 16]
```diff
@@ -201,39 +205,43 @@ export class EnsController extends BaseControllerV2<
...
    */
-  set(chainId: Hex, ensName: string, address: string | null): boolean {
+  set(
+    caipChainId: CaipChainId,
+    ensName: string,
+    address: string | null,
+  ): boolean {
     if (
-      !Number.isInteger(Number.parseInt(chainId, 10)) ||
+      !isEthCaipChainId(caipChainId) || // is this right?
```
Comment (adonesky1) [inline packages/ens-controller/src/EnsController.ts:238]: looks right to me!

**Comment by jiexi** [comment]
> Needs to be a breaking changelog entry indicating that SupportedTokenDetectionNetworks is now CAIP formatted

`SupportedTokenDetectionNetworks` isn't exported, but it is used by `isTokenDetectionSupportedForNetwork` which is exported. Will add an entry for that

File: packages/controller-utils/src/caip.test.ts (hunk 12 above)
Comment (jiexi) [reply packages/controller-utils/src/caip.test.ts:21]: it will throw `Invalid chain ID "${chainId}"`

File: packages/controller-utils/src/caip.test.ts (hunk 13 above)
Comment (jiexi) [reply packages/controller-utils/src/caip.test.ts:94]: hard to say. Currently in extension, the pattern i see is that we check for NaN in if guard statements after attempting to parseInt. Throwing would change this pattern from

```
const chainIdInt = parseInt(chainId, 10)

if (Number.isNaN(chainIdInt)) {
  // exit if NaN
  return
}

doSomething(chainIdInt)
```

to

```
let chainIdInt

try {
chainIdInt = parseEthCaipChainIdInt(caipChainId)
} catch (err) {
    // exit if NaN
}

doSomething(chainIdInt)
```

it seems like we're deciding if we should treat NaN as an exception or not

File: packages/controller-utils/src/caip.ts (hunk 14 above)
Comment (jiexi) [reply packages/controller-utils/src/caip.ts:37]: yeah, makes sense to me

File: packages/controller-utils/src/constants.ts (hunk 15 above)
Comment (jiexi) [reply packages/controller-utils/src/constants.ts:140]: correct. unrelated to CAIP, but wanted to help disambiguate NetworkId and ChainId more by this naming. Happy to revert though

File: packages/controller-utils/src/caip.test.ts (hunk 12 above)
Comment (jiexi) [reply packages/controller-utils/src/caip.test.ts:21]: changing this to just return the caip chain id

File: packages/controller-utils/src/caip.ts (hunk 8 above)
Comment (jiexi) [reply packages/controller-utils/src/caip.ts:72]: Regardless of whether or not it is safe, we are already doing this conversion in a few places. From what I've seen, the preference in our usage is Hex -> Decimal (string) -> Decimal (number). We generally only do the conversion from Decimal string to number when we are comparing against chainId number json values returned by some outside API

File: packages/controller-utils/src/caip.test.ts (hunk 7 above)
Comment (jiexi) [reply packages/controller-utils/src/caip.test.ts:91]: but it's technically a valid CAIP chainId format, but invalid reference for the namespace

File: packages/controller-utils/src/caip.ts (hunk 8 above)
Comment (jiexi) [reply packages/controller-utils/src/caip.ts]: Renamed the helpers

**Comment by jiexi** [comment]
@metamaskbot publish-preview

File: packages/gas-fee-controller/package.json [hunk 17]
```diff
@@ -33,7 +33,7 @@
     "@metamask/controller-utils": "workspace:^",
     "@metamask/eth-query": "^3.0.1",
     "@metamask/network-controller": "workspace:^",
-    "@metamask/utils": "^6.2.0",
+    "@metamask/utils": "^7.1.0",
```
Comment (mcmire) [inline packages/gas-fee-controller/package.json:36]: Should we note this in the changelog?

File: packages/keyring-controller/package.json [hunk 18]
```diff
@@ -34,7 +34,7 @@
     "@metamask/eth-keyring-controller": "^13.0.0",
     "@metamask/message-manager": "workspace:^",
     "@metamask/preferences-controller": "workspace:^",
-    "@metamask/utils": "^6.2.0",
+    "@metamask/utils": "^7.1.0",
```
Comment (mcmire) [inline packages/keyring-controller/package.json:37]: Should we note this in the changelog?

File: packages/message-manager/package.json [hunk 19]
```diff
@@ -32,7 +32,7 @@
     "@metamask/base-controller": "workspace:^",
     "@metamask/controller-utils": "workspace:^",
     "@metamask/eth-sig-util": "^6.0.0",
-    "@metamask/utils": "^6.2.0",
+    "@metamask/utils": "^7.1.0",
```
Comment (mcmire) [inline packages/message-manager/package.json:35]: Should we note this in the changelog?

File: packages/network-controller/package.json [hunk 20]
```diff
@@ -36,7 +36,7 @@
     "@metamask/eth-json-rpc-provider": "^1.0.0",
     "@metamask/eth-query": "^3.0.1",
     "@metamask/swappable-obj-proxy": "^2.1.0",
-    "@metamask/utils": "^6.2.0",
+    "@metamask/utils": "^7.1.0",
```
Comment (mcmire) [inline packages/network-controller/package.json:39]: Should we note this in the changelog?

File: packages/network-controller/tests/NetworkController.test.ts [hunk 21]
```diff
@@ -7309,7 +7340,7 @@ function lookupNetworkTests({
   operation: (controller: NetworkController) => Promise<void>;
 }) {
   describe('if the network ID and network details requests resolve successfully', () => {
-    const validNetworkIds = [12345, '12345', toHex(12345)];
+    const validNetworkIds = [12345, '12345', toHex(12345)]; // ?
```
Comment (mcmire) [inline packages/network-controller/tests/NetworkController.test.ts:7343]: Some chains report their network IDs in hex format: https://github.com/MetaMask/metamask-extension/issues/19151

File: packages/notification-controller/package.json [hunk 22]
```diff
@@ -30,7 +30,7 @@
   },
   "dependencies": {
     "@metamask/base-controller": "workspace:^",
-    "@metamask/utils": "^6.2.0",
+    "@metamask/utils": "^7.1.0",
```
Comment (mcmire) [inline packages/notification-controller/package.json:33]: Should we note this in the changelog?

File: packages/permission-controller/package.json [hunk 23]
```diff
@@ -32,7 +32,7 @@
     "@metamask/approval-controller": "workspace:^",
     "@metamask/base-controller": "workspace:^",
     "@metamask/controller-utils": "workspace:^",
-    "@metamask/utils": "^6.2.0",
+    "@metamask/utils": "^7.1.0",
```
Comment (mcmire) [inline packages/permission-controller/package.json:35]: Should we note this in the changelog?

File: packages/signature-controller/package.json [hunk 24]
```diff
@@ -33,7 +33,7 @@
     "@metamask/base-controller": "workspace:^",
     "@metamask/controller-utils": "workspace:^",
     "@metamask/message-manager": "workspace:^",
-    "@metamask/utils": "^6.2.0",
+    "@metamask/utils": "^7.1.0",
```
Comment (mcmire) [inline packages/signature-controller/package.json:36]: Should we note this in the changelog?

File: packages/transaction-controller/package.json [hunk 25]
```diff
@@ -36,7 +36,7 @@
     "@metamask/controller-utils": "workspace:^",
     "@metamask/eth-query": "^3.0.1",
     "@metamask/network-controller": "workspace:^",
-    "@metamask/utils": "^6.2.0",
+    "@metamask/utils": "^7.1.0",
```
Comment (mcmire) [inline packages/transaction-controller/package.json:39]: Should we note this in the changelog?

File: packages/transaction-controller/src/TransactionController.ts [hunk 26]
```diff
@@ -1061,11 +1065,12 @@ export class TransactionController extends BaseController<
   ): TransactionMeta[] {
     const nonceNetworkSet = new Set();
     const txsToKeep = transactions.reverse().filter((tx) => {
-      const { chainId, networkID, status, transaction, time } = tx;
+      const { caipChainId, networkID, status, transaction, time } = tx;
       if (transaction) {
-        const key = `${transaction.nonce}-${
-          chainId ? convertHexToDecimal(chainId) : networkID
-        }-${new Date(time).toDateString()}`;
+        const networkChainId = caipChainId || networkID; // is this right?
```
Comment (mcmire) [inline packages/transaction-controller/src/TransactionController.ts:1070]: It might be good to keep the same behavior and convert the CAIP chain ID to decimal so that keys for existing transactions are the same. Otherwise transactions that are already in state might get accidentally dropped.

Comment (mcmire) [reply packages/transaction-controller/src/TransactionController.ts:1070]: (Side note: it might be good to ask these questions in GitHub comments rather than code comments so they don't get accidentally committed.)

**Review by mcmire — COMMENTED** [review]
Did a pass over this. I noticed some questions you asked in code comments as I was going over this and it's possible I didn't catch all of them. I will do another pass later.

One thing that stood out to me, though, was that we are renaming `chainId` across various pieces of state and configuration, and I wonder whether that is a good idea, or whether it would just be enough to change the names of the types. My concern is that this would spawn a _ton_ of changes across extension and mobile when upgrading to newer versions of these packages (changing the type will spawn enough changes as it is). The more changes we have to make, the more we risk introducing a bug.

So, I'm curious what the reason was to do this. Was it just to hammer home the fact that this is a CAIP chain ID and not a "legacy" chain ID? Was it to ensure that consumers are using the correct chain ID when they upgrade? My understanding is that we are moving to a world in which chain ID _always_ means CAIP-2 chain ID, so when that happens, eventually the "Caip" part of the name will become redundant. But, are we at a point where we can't guarantee that a chain ID is always a CAIP-2 chain ID?

**Comment by jiexi** [comment]
Postponed. Not sure if/when will be picked back up.
